lib/OpenQA/Worker.pm

```diff
@@ -14,108 +14,341 @@
 # with this program; if not, see <http://www.gnu.org/licenses/>.
 
 package OpenQA::Worker;
+use Mojo::Base -base;
 
-use 5.018;
-use strict;
-use warnings;
-
-use Mojolicious::Lite;
-use Mojo::Server::Daemon;
-use Mojo::IOLoop;
+use POSIX 'uname';
+use Fcntl;
+use File::Path qw(make_path remove_tree);
 use File::Spec::Functions 'catdir';
+use Mojo::IOLoop;
 use Mojo::File 'path';
 use Try::Tiny;
+use Scalar::Util 'looks_like_number';
+use OpenQA::Constants qw(WEBSOCKET_API_VERSION MAX_TIMER MIN_TIMER);
 use OpenQA::Client;
-use OpenQA::Utils qw(log_error log_info log_debug);
-use OpenQA::Worker::Common;
-use OpenQA::Worker::Commands;
-use OpenQA::Worker::Pool qw(lockit clean_pool);
-use OpenQA::Worker::Jobs;
+use OpenQA::Utils qw(log_error log_info log_debug add_log_channel remove_log_channel);
+use OpenQA::Worker::WebUIConnection;
+use OpenQA::Worker::Settings;
+use OpenQA::Worker::Job;
 use OpenQA::Setup;
 
-sub init {
-    my ($host_settings, $options) = @_;
-    $instance  = $options->{instance} if defined $options->{instance};
-    $pooldir   = $OpenQA::Utils::prjdir . '/pool/' . $instance;
-    $nocleanup = $options->{"no-cleanup"};
+has 'instance_number';
+has 'pool_directory';
+has 'no_cleanup';
+has 'app';
+has 'settings';
+has 'clients_by_webui_host';
+has 'current_webui_host';
+has 'current_job';
+has 'current_error';
+has 'worker_hostname';
+has 'isotovideo_interface_version';
+
+sub new {
+    my ($class, $cli_options) = @_;
+
+    # determine uname info
+    my ($sysname, $hostname, $release, $version, $machine) = POSIX::uname();
 
-    my $logdir = $ENV{OPENQA_WORKER_LOGDIR} // $worker_settings->{LOG_DIR};
-    my $app    = OpenQA::Setup->new(
+    # determine instance number
+    my $instance_number = $cli_options->{instance};
+    if (!defined $instance_number) {
+        die 'no instance number specified';
+    }
+    if (!looks_like_number($instance_number)) {
+        die "the specified instance number \"$instance_number\" is no number";
+    }
+
+    # determine settings and create app
+    my $settings = OpenQA::Worker::Settings->new($instance_number, $cli_options);
+    my $app      = OpenQA::Setup->new(
         mode     => 'production',
         log_name => 'worker',
-        instance => $instance,
-        log_dir  => $logdir
+        instance => $instance_number,
+    );
+    $settings->apply_to_app($app);
+
+    # setup the isotovideo engine
+    # FIXME: Get rid of the concept of engines in the worker.
```

Comment:
There is not disabled code. This is just a hint about an improvement which I didn't do because the PR would become too big. What let's you think that each FIXME/TODO comment is related to disabled code?

Reply:
sorry, wrong github review template, should be "use tickets for tracking tasks"

Reply:
I actually like hints about flawed places in our code. This makes it easier for people to know which patterns should better *not* be copied.